src/components/forms/ClusteredDiskSizeSelector.tsx

```diff
@@ -0,0 +1,231 @@
+
+  const [isSpecific, setIsSpecific] = useState<boolean | null>(
+    isDefaultSpecific,
+  );
+  const firstValue = Object.values(sizeValues ?? {})[0];
+
+  const limit = (val: string) => {
+    return (
+      parseMemoryLimit(val) ?? {
+        value: 1,
+        unit: BYTES_UNITS.GIB,
+      }
+    );
+  };
+
+  const getMemUnitOptions = () => {
+    return Object.values(BYTES_UNITS).map((unit) => ({
+      label: unit,
+      value: unit,
+    }));
+  };
+
+  useEffect(() => {
+    const rawValues = Object.values(sizeValues ?? {});
+    if (isSpecific === null && rawValues.length > 0) {
+      const newDefaultSpecific = rawValues.some(
+        (item) => item !== rawValues[0],
+      );
+      setIsSpecific(newDefaultSpecific);
+    }
+  }, [isSpecific, sizeValues]);
+
+  const setValueForAllMembers = (value: string) => {
+    const update: ClusterSpecificValues = {};
+    memberNames.forEach((member) => (update[member] = value));
+    setMemoryLimit(update);
+  };
+
+  const setValueForMember = (value: string, member: string) => {
+    const update = {
+      ...sizeValues,
+      [member]: value,
+    };
+    setMemoryLimit(update);
+  };
+
+  return (
+    <div className="u-sv3">
+      <Label forId="poolSizePerClusterMember">Size</Label>
+      {canToggleSpecific && !isReadOnly && (
+        <CheckboxInput
+          id={`${id}-same-for-all-toggle`}
+          label="Same for all cluster members"
+          checked={!isSpecific}
+          onChange={() => {
+            setIsSpecific((val) => !val);
+          }}
+        />
+      )}
+      {isSpecific && (
+        <div className="cluster-specific-input">
+          {memberNames.map((item) => {
+            const activeValue = sizeValues?.[item];
+
+            return (
+              <Fragment key={item}>
+                <div className="cluster-specific-member">
+                  <ResourceLink
+                    type="cluster-member"
+                    value={item}
+                    to={clusterMemberLinkTarget(item)}
+                  />
+                </div>
+                <div className="cluster-specific-value">
+                  {isReadOnly ? (
+                    <>
+                      {activeValue}
+                      <FormEditButton toggleReadOnly={toggleReadOnly} />
+                    </>
+                  ) : (
+                    <div className="memory-limit-with-unit">
+                      <Input
+                        id={
+                          memberNames.indexOf(item) === 0 ? id : `${id}-${item}`
+                        }
+                        name="limits_disk"
+                        type="number"
+                        min="0"
+                        step="Any"
+                        placeholder="Enter value"
+                        onChange={(e) =>
+                          setValueForMember(
+                            e.target.value + limit(activeValue || "").unit,
```

Comment:
I think better to refactor the definition as `const activeValue = sizeValues?.[item] ?? "";` and then we can simplify all the use cases of it. Also the ones below.
```suggestion
                            e.target.value + limit(activeValue).unit,
```